_posts/2019-11-27-gettingstarted-2020/getting-started-2020.Rmd

```diff
@@ -0,0 +1,144 @@
+</aside>
+
+The tutorials just mentioned are all nice introductions explaining code as well as concepts. What if you're familiar with the basic procedure and just need a quick reminder (or: something to quickly copy-paste from)? The ideal document to consult for those purposes is the [Overview](https://tensorflow.rstudio.com/tutorials/beginners/).
+
+Now -- knowledge how to build models is fine, but as in data science overall, there is no modeling without data.
+
+### Data ingestion and preprocessing
+
+Two detailed, end-to-end tutorials show how to load [csv data](https://tensorflow.rstudio.com/tutorials/beginners/load/load_csv/) and 
+[images](https://tensorflow.rstudio.com/tutorials/beginners/load/load_image/), respectively.
+
+In current Keras, two mechanisms are central to data preparation. One is the use of [tfdatasets pipelines](https://tensorflow.rstudio.com/guide/tfdatasets/introduction/). `tfdatasets` lets you load data in a streaming fashion (batch-by-batch), optionally applying transformations as you go. The other handy device here is [feature specs](https://tensorflow.rstudio.com/guide/tfdatasets/feature_spec/) resp. [feature columns](https://tensorflow.rstudio.com/guide/tfdatasets/feature_columns/). Together with a matching Keras layer, these allow for transforming the input data without having to think about what the new format will mean to Keras.
+
+While there are other types of data not discussed in the docs, the principles -- pre-processing pipelines and feature extraction -- generalize.
+
+### Model saving
+
+The best-performing model is of little use if ephemeral. Straightforward ways of saving Keras models are explained in a dedicated [tutorial](https://tensorflow.rstudio.com/tutorials/beginners/basic-ml/tutorial_save_and_restore/). 
+
+<aside>
+Advanced users: Additional options exist; see the [tutorial on checkpoints](https://tensorflow.rstudio.com/guide/saving/checkpoints/).
+</aside>
+
+And unless one's just tinkering around, the question will often be: How can I deploy my model?
+There is a complete new section on [deployment](https://tensorflow.rstudio.com/deploy/), featuring options like `plumber`, Shiny, TensorFlow Serving and RStudio Connect.
+
+<aside>
+Check out the new section on [deployment options](https://tensorflow.rstudio.com/deploy/).
+</aside>
+
+After this workflow-oriented run-through, let's see about different types of data you might want to model.
+
+## Neural networks for different kinds of data
+
+No introduction to deep learning without image classification. The "Fashion MNIST" classification tutorial mentioned in the beginning is a good introduction, but it uses a fully connected neural network to make it easy to remain focused on the overall approach. Standard models for image recognition, however, are commonly based on a convolutional architecture. [Here](https://tensorflow.rstudio.com/tutorials/advanced/images/cnn/) is a nice introductory tutorial.
```

Comment:
```suggestion
No introduction to deep learning is complete without image classification. The "Fashion MNIST" classification tutorial mentioned in the beginning is a good introduction, but it uses a fully connected neural network to make it easy to remain focused on the overall approach. Standard models for image recognition, however, are commonly based on a convolutional architecture. [Here](https://tensorflow.rstudio.com/tutorials/advanced/images/cnn/) is a nice introductory tutorial.
```

Reply:
You don't have to use "is complete" but you need some sort of condition there.

Reply:
oh yeah, much better...